src/command.rs

```diff
@@ -170,85 +209,54 @@ impl<'s> Command<'s> {
     /// [`null`]: struct.Stdio.html#method.null
     /// [`piped`]: struct.Stdio.html#method.piped
     pub fn stderr<T: Into<Stdio>>(&mut self, cfg: T) -> &mut Self {
-        self.builder.stderr(cfg);
-        self.stderr_set = true;
+        match &mut self.imp {
+            CommandImp::ProcessImpl(imp) => {
+                imp.stderr(cfg.into());
+            }
+
+            #[cfg(feature = "mux_client")]
+            CommandImp::MuxClientImpl(imp) => {
+                imp.stderr(cfg.into());
+            }
+        };
         self
     }
 
     /// Executes the remote command without waiting for it, returning a handle to it instead.
     ///
     /// By default, stdin is empty, and stdout and stderr are discarded.
-    pub fn spawn(&mut self) -> Result<RemoteChild<'s>, Error> {
-        // Make defaults match our defaults.
-        if !self.stdin_set {
-            self.builder.stdin(Stdio::null());
-        }
-        if !self.stdout_set {
-            self.builder.stdout(Stdio::null());
-        }
-        if !self.stderr_set {
-            self.builder.stderr(Stdio::null());
-        }
-        // Then launch!
-        let child = self.builder.spawn().map_err(Error::Ssh)?;
+    ///
+    /// After this function, stdin, stdout and stderr is reset.
```

Comment:
That is behavior exclusive to `mux_client_impl::Command`, for not resetting it will potentially cause the remote process to hang forever if user passed in an arbitary fd and forget to close it before calling `RemoteChild::wait`.

Reply:
Or I can change that to reset stdin, stdout and stderr only if an arbitary fd is passed in.

Reply:
OK, this is fixed.

Now `mux_client_impl::Command::spawn` do not touch stdin/stdout/stderr.